frontend/frontend/protocols/ssh.py

```diff
@@ -124,40 +214,47 @@ def listen(self) -> None:
 
         try:
             sock.listen(self._max_unaccepted_connections)
-            # todo Only accepts one client for now
-            client, addr = sock.accept()
-            transport = paramiko.Transport(client)
         except Exception as exc:
             print(f"Failed to accept connection\nError{exc}")
             raise
 
-        print(f"Client {addr[0]}:{addr[1]} connected")
+        # This provides type hinting
+        instance_list:  List[ConnectionHandler]
+        instance_list = []
 
-        if not transport.load_server_moduli():
-            print("Could not load moduli")
+        sock.settimeout(socket_timeout)
+        while True:
+            self._lock.acquire()
+            if self._terminate:
+                self._lock.release()
+                break
+            self._lock.release()
+
+            # Try accepting connections
+            try:
+                client, addr = sock.accept()
+                transport = paramiko.Transport(client)
+            except Exception as exc:
+                # todo log
+                continue
 
-        # Negotiate a new SSH session
-        transport.add_server_key(self._host_key)
-        server = Server(self._usernames, self._passwords)
-        transport.start_server(server=server)
+            session = logger.begin_ssh_session(
+                src_address=ip_address(addr[0]),
```

Comment:
I see a lot of argument passing like this. Wouldn't you benefit from containing all 4 inside its own class instead?